pysmt/parsing.py

```diff
@@ -0,0 +1,435 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2014 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import re
+from fractions import Fraction
+
+from pysmt.shortcuts import get_env
```

Comment:
This should not call shortcuts, but use the functions from environment instead. This was done to avoid circular dependencies, and make ```shortcuts``` the top-level module for everything "outside" pySMT.